Makefile

```diff
@@ -13,7 +13,13 @@
 # limitations under the License.
 
 COMPILER?=g++
+
+GCC_VER_GTE53 := $(shell echo `gcc -dumpversion | cut -f1-2 -d.` \>= 5.3 | sed -e 's/\./*100+/g' | bc )
```

Comment:
This line hardcodes use of the "gcc" executable. Your build is now failing on the gcc-5 build because the executable that's being used is called "gcc-5", not "gcc", and your check gets the wrong result.

How about using $(COMPILER) from two lines up, that's set from the Travis file if it's not just gcc/g++ ?